lib/factory_bot/registry.rb

```diff
@@ -1,4 +1,22 @@
 module FactoryBot
+  # In Ruby <= 2.5 it is necessary to monkey patch
+  # the key error class because it does not expose
+  # receiver and key attributes.
+  # See https://bugs.ruby-lang.org/issues/14313
+  module KeyErrorExtension
+    attr_reader :receiver, :key
+
+    def initialize(msg, key: nil, receiver: nil)
+      @key = key
+      @receiver = receiver
+      super msg
+    end
+  end
+
+  if Gem::Version.new(RUBY_VERSION) < Gem::Version.new("2.6")
```

Comment:
I don't think this would work for jruby or other implementations, but I don't know if factory_bot currently works at all for any other Ruby implementations. We stopped running CI for jruby in https://github.com/thoughtbot/factory_bot/pull/1018, and are rbx build hasn't been passing lately.

Reply:
Just verified with ``rubinius 3.107 `` and ``jruby 9.2.0.0 (2.5.0)``, works for both. ``RUBY_VERSION`` is in both implementations the actual Ruby version without the platform.